app/uk/gov/hmrc/exports/services/reversemapping/declaration/DeclarationXmlParser.scala

```diff
@@ -44,4 +45,8 @@ object DeclarationXmlParser {
       insideOut(list, Nil)
     }
   }
+
+  implicit class OptionalString(nodeSeq: NodeSeq) {
+    def toStringOption: Option[String] = StringOption(nodeSeq.text)
+  }
```

Comment:
Maybe I'm wrong, but won't this create a new OptionalString instance for every NodeSeq in scope (even for those NodeSeqs that do not represent strings)? If so seems a bit wasteful no?

Reply:
Do you mean at runtime? I don't think so. Only for the **types** in scope. Do not forget that the class is just a template at compile time.